src/odemis/gui/util/img.py

```diff
@@ -1414,6 +1414,106 @@ def chronogram_to_export_data(proj, raw, vp=None):
         scale_y_draw = numpy.append(extend[:SMALL_SCALE_WIDTH, :], scale_y_draw, axis=0)
         data_with_legend = numpy.append(scale_y_draw, data_with_legend, axis=1)
 
+        # Draw right vertical scale legend
+        scale_y_draw = draw_scale(range_y, (SMALL_SCALE_WIDTH, client_size[1]), wx.VERTICAL,
+                              tick_spacing, text_colour, "cts", font_size, None,
+                              mirror=True)
+
+        # Extend y scale bar to fit the height of the bar pl
+        # ot with the x scale bars attached
+        scale_y_draw = numpy.append(scale_y_draw, extend[:, :SMALL_SCALE_WIDTH], axis=0)
+        scale_y_draw = numpy.append(extend[:SMALL_SCALE_WIDTH, :SMALL_SCALE_WIDTH], scale_y_draw, axis=0)
+        data_with_legend = numpy.append(data_with_legend, scale_y_draw, axis=1)
+
+        spec_plot = model.DataArray(data_with_legend)
+        spec_plot.metadata[model.MD_DIMS] = 'YXC'
+        return spec_plot
+
+
+def theta_to_export_data(proj, raw, vp=None):
+    """
+    Creates either the raw or the representation as shown in the viewport in the GUI of the theta data plot for export.
+    :param proj: (SinglePointAngularProjection) A theta projection.
+    :param raw: (boolean) If True, returns raw representation.
+    :param vp: (Viewport or None) The viewport selected for export to get the data displayed
+               in the viewport.
+    :returns: (model.DataArray) The data array to export.
+    """
+    if not isinstance(proj, SinglePointAngularProjection):
+        raise ValueError("Trying to export an angle spectrum of an invalid projection")
+
+    if raw:  # csv
+        data = proj.projectAsRaw()
+        if data is None:
+            raise LookupError("No pixel selected to pick a theta")
+        data.metadata[model.MD_ACQ_TYPE] = model.MD_AT_EK
+        return data
+    else:  # tiff, png
+        spec = proj.image.value
+        if spec is None:
+            raise LookupError("No pixel selected to pick a theta")
+        angle_range, unit = spectrum.get_angle_range(spec)
+
+        def remove_nan(x, y):
+            """Removes NaN values from the angle list and updates the spectrum to have the same length"""
+            filtered = [(xv, yv) for xv, yv in zip(x, y) if not math.isnan(xv)]
+            return [f[0] for f in filtered], [f[1] for f in filtered]
```

Comment:
I think that eventually this could be done in the stream, on the .calibrated for the StaticStream, and possibly in the projection of the SettingsStream. Then this code wouldn't been needed here or in the Viewport.